### PR TITLE
Add synchronous message consuming functionality to RabbitMQ connector

### PR DESCRIPTION
**Pull request by aashikam**
AMQP Reference: https://www.rabbitmq.com/amqp-0-9-1-reference.html#basic.get

## Samples
```ballerina
import ballerina/io;
import ballerina/rabbitmq;

public function main() {
    rabbitmq:Connection connection = new({ host: "localhost", port: 5672 });

    rabbitmq:Channel newChannel1 = new(connection);

    var queueResult = newChannel1->queueDeclare({ queueName: "MyQueue" });
    if (queueResult is error) {
        io:println("An error occurred while creating the MyQueue1 queue.");
    }
    var sendResult = newChannel1->basicPublish("Hello from Ballerina", "MyQueue");
    if (sendResult is error) {
        io:println("An error occurred while sending the message to " +
                 "MyQueue1 using newChannel1.");
    }

    var getResult = newChannel1->basicGet("MyQueue", rabbitmq:AUTO_ACK);
    if (getResult is rabbitmq:Message) {
        var message = getResult.getTextContent();
        if (message is string) {
            io:println(message);
        }
    }
}
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            GetResponse response = channel.basicGet(queueName, autoAck);
+            return getMessageObjectValue(response.getBody(), response.getEnvelope().getDeliveryTag(),
+                    response.getProps(), channel, autoAck);
+        } catch (IOException e) {
+            return RabbitMQUtils.returnErrorValue("Error occurred while retrieving the message: " +
+                    e.getMessage());
+        }
+    }
+
+    private static ObjectValue getMessageObjectValue(byte[] message, long deliveryTag, AMQP.BasicProperties properties,
```
Comment (riyafa) [inline stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java]: How about renaming to make the intent clear?
```suggestion
    private static ObjectValue createAndPopulateMessageObjectValue(byte[] message, long deliveryTag, AMQP.BasicProperties properties,
```

Comment (aashikam) [reply stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java]: As this method is a private method used only by this class it is clear what it does in this context, without making the method name unnecessarily long. But if it is still **confusing** we can change the method name.

Comment (riyafa) [reply stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java]: Quoting from clean code:

> Don’t be afraid to make a name long. A long descriptive name is better than a shortenigmatic name. A long descriptive name is better than a long descriptive comment. 

This is true specially for private functions. If you think changing function name doesn't add value we can leave it.

Comment (aashikam) [reply stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java]: Resolving the method name.

File: stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    public static Object basicGet(Strand strand, ObjectValue channelObjectValue, String queueName, Object ackMode) {
+        Channel channel = (Channel) channelObjectValue.getNativeData(RabbitMQConstants.CHANNEL_NATIVE_OBJECT);
+        boolean autoAck = false;
+        if (ackMode.toString().equals("auto")) {
+            autoAck = true;
+        }
+        try {
+            GetResponse response = channel.basicGet(queueName, autoAck);
+            return createAndPopulateMessageObjectValue(response.getBody(), response.getEnvelope().getDeliveryTag(),
```
Comment (wggihan) [inline stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java]: Shall we pass the response object as an parameter and reduce the parameter count?

Comment (aashikam) [reply stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicGet.java]: Refactored the method.

**Pull request merged**